src/foam/nanos/crunch/lite/ruler/CapableStatusChangeAdjustApprovalsRuleAction.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.lite.ruler',
+  name: 'CapableStatusChangeAdjustApprovalsRuleAction',
+
+  documentation: `
+    TODO:
+  `,
+
+  javaImports: [
+    'foam.core.ContextAwareAgent',
+    'foam.core.FObject',
+    'foam.core.X',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.nanos.auth.User',
+    'foam.util.SafetyUtil',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.approval.ApprovalRequest',
+    'foam.nanos.approval.Approvable',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.lite.Capable',
+    'foam.nanos.crunch.lite.CapablePayload',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.approval.ApprovalStatus',
+    'foam.nanos.ruler.Operations',
+    'foam.nanos.auth.Subject',
+    'java.util.ArrayList',
+    'java.util.List',
+    'java.util.Map',
+    'java.util.HashMap'
+  ],
+
+  implements: ['foam.nanos.ruler.RuleAction'],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        User user = ((Subject) x.get("subject")).getUser();
+
+        Logger logger = (Logger) x.get("logger");
+
+        DAO capabilityDAO = (DAO) x.get("capabilityDAO");
+
+        Capable capableNewObj = (Capable) obj;
+        Capable capableOldObj = (Capable) oldObj;
+
+        CapablePayload[] newCapablePayloads = capableNewObj.getCapablePayloads();
+        CapablePayload[] oldCapablePayloads = capableOldObj.getCapablePayloads();
+
+        List<CapablePayload> updatedApprovalPayloads = new ArrayList<>();
+
+        if ( newCapablePayloads.length != oldCapablePayloads.length ){
```

Comment:
You have this as an after rule, so I think if this gets run twice, the second time it will accept it.

Reply:
Resolved by CapableDAO cleanup